.gitlab-ci.yml

```diff
@@ -1,20 +1,21 @@
-image: docker:latest
-
 stages:
   - build
   - test
   - status
 
 .build_template: &build_definition
   stage: build
+  image:
+    name: gcr.io/kaniko-project/executor:debug
+    entrypoint: [""]
   before_script:
-    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
-    - docker pull $CI_REGISTRY/$CI_PROJECT_PATH/$CI_JOB_NAME || true
+    - echo "{\"auths\":{\"$CI_REGISTRY\":{\"username\":\"$CI_REGISTRY_USER\",\"password\":\"$CI_REGISTRY_PASSWORD\"}}}" > /kaniko/.docker/config.json
```

Comment:
if I'm evil and open a PR after this is merged, couldn't I just read out this file?

Reply:
No, these credentials are invalidated once the job completes. Also, you could already read the environment variables before.